src/Core/Domain/Product/SpecificPrice/Command/AddProductSpecificPriceCommand.php

```diff
@@ -283,15 +309,19 @@ public function getCurrencyId(): ?int
      */
     public function setCurrencyId(int $currencyId): self
     {
-        $this->currencyId = $currencyId;
+        if (NoCurrencyId::NO_CURRENCY_ID === $currencyId) {
+            $this->currencyId = new NoCurrencyId();
+        } else {
+            $this->currencyId = new CurrencyId($currencyId);
+        }
 
         return $this;
     }
 
     /**
-     * @return int|null
+     * @return int
      */
-    public function getCountryId(): ?int
+    public function getCountryId(): int
```

Comment:
yes it should, but as I've explained in early stages of this PR - country and currency (i think) has already been used in many places where it does allow 0 in VO so to refactor them in to NoIdInterface convention would require alot of time and effort (and definitely additional PR). So I left it just like this. (I could also use here CountryId and CurrencyId value objects as those accepts 0, but I think it would be even more confusing)